scp/cc/cpio/client_providers/private_key_fetcher_provider/src/azure/azure_private_key_fetcher_provider_utils.cc

```diff
@@ -0,0 +1,39 @@
+/*
+ * Portions Copyright (c) Microsoft Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "azure_private_key_fetcher_provider_utils.h"
+
+#include "azure/attestation/json_attestation_report.h"
+
+
+using google::scp::core::HttpRequest;
+using google::scp::core::Uri;
+using google::scp::core::HttpMethod;
+
+namespace google::scp::cpio::client_providers {
+
+void AzurePrivateKeyFetchingClientUtils::CreateHttpRequest(
```

Comment:
**[Comment for actual PR]**
We have TODOs for future PRs in this function:
- Handle `private_key_cache_ttl_seconds` option.
- Allow to specify `key_id` for private key API